Dockerfiles/gadget-builder.Dockerfile

```diff
@@ -22,30 +26,50 @@ RUN \
 	tar -C /usr/local/bin -xzf bpftool-${BPFTOOL_VERSION}-${ARCH}.tar.gz && \
 	chmod +x /usr/local/bin/bpftool
 
-# We use Go version >1.24 here for the WASM layers, because it is the first version to support exporting functions
-FROM golang:1.24.2@sha256:30baaea08c5d1e858329c50f29fe381e9b7d7bced11a0f5f1f69a1504cdfbf5e
+FROM debian:bookworm-slim@sha256:4b50eb66f977b4062683ff434ef18ac191da862dbe966961bc11990cf5791a8d
 ARG CLANG_LLVM_VERSION
+ARG GOLANG_VERSION
 # libc-dev is needed for various headers, among others
 # /usr/include/arch-linux-gnu/asm/types.h.
 # We make clang aware of these files through CFLAGS in Makefile.
-# lsb-release wget software-properties-common gnupg are needed by llvm.sh script
+# wget is needed to download the LLVM key and Golang tarball
+# lsb-release software-properties-common is needed to add the LLVM repository
 # xz-utils is needed by btfgen makefile
+# make and git is needed for make ebpf-objects and make clang-format
+# clang-format is needed for make clang-format
 RUN apt-get update \
-	&& apt-get install -y libc-dev lsb-release wget gnupg xz-utils software-properties-common
+	&& apt-get install -y --no-install-recommends libc-dev lsb-release wget xz-utils software-properties-common make git
 
-# Install clang 15
-RUN wget https://apt.llvm.org/llvm.sh && chmod +x llvm.sh && ./llvm.sh $CLANG_LLVM_VERSION all \
+# Install clang
+# Add the keys and repository for the LLVM packages
+RUN wget -qO- https://apt.llvm.org/llvm-snapshot.gpg.key | tee /etc/apt/trusted.gpg.d/apt.llvm.org.asc
+# We need to call add-apt-repository twice in debian-bookworm because of a bug: https://github.com/llvm/llvm-project/issues/62475#issuecomment-1579252282
+RUN codename=$(lsb_release -cs); add-apt-repository -y "deb http://apt.llvm.org/${codename}/ llvm-toolchain-${codename}-${CLANG_LLVM_VERSION} main" \
+	&& add-apt-repository -y "deb http://apt.llvm.org/${codename}/ llvm-toolchain-${codename}-${CLANG_LLVM_VERSION} main"
+RUN apt-get update \
+	&& apt-get install -y --no-install-recommends clang-$CLANG_LLVM_VERSION llvm-$CLANG_LLVM_VERSION clang-format-$CLANG_LLVM_VERSION \
 	&& update-alternatives --install /usr/local/bin/llvm-strip llvm-strip $(which llvm-strip-$CLANG_LLVM_VERSION) 100 \
 	&& update-alternatives --install /usr/local/bin/clang clang $(which clang-$CLANG_LLVM_VERSION) 100 \
 	&& update-alternatives --install /usr/local/bin/clang-format clang-format $(which clang-format-$CLANG_LLVM_VERSION) 100
 
+# Install golang
```

Comment:
I remember there was an issue with golang dockerfiles regarding toolchain upgrading golang version. Perhaps we should check if we need some environment variables to avoid this. See; https://github.com/docker-library/golang/blob/master/1.24/bookworm/Dockerfile#L121